builder/config_reader_test.go

```diff
@@ -108,30 +108,31 @@ func testConfig(t *testing.T, when spec.G, it spec.S) {
 				})
 			})
 
-			when("'uri' is misspelled as url", func() {
+			when("unknown buildpack key is present", func() {
 				it.Before(func() {
 					h.AssertNil(t, ioutil.WriteFile(builderConfigPath, []byte(`
-[buildpack]
+[[buildpacks]]
```

Comment:
It took me a long time to figure this out! In builder.toml, `buildpacks` (plural) is a table name, and it is a list - so it needs double brackets `[[]]`. (This is in contrast to package.toml, where `buildpack` (singular) is a table name, and there is only one.) 

For builder.toml, when we are testing invalid key, we need to provide a valid table - otherwise the error message will be specific to the table and not the key.

Reply:
Okay, it makes sense now! I had been otherwise searching through the validator for potential bugs!